tests/std/tests/P0896R4_stream_iterators/test.cpp

```diff
@@ -0,0 +1,125 @@
+        STATIC_ASSERT(is_nothrow_constructible_v<I, default_sentinel_t>);
+    }
+
+    assert(i == default_sentinel);
+    STATIC_ASSERT(noexcept(i == default_sentinel));
+    assert(default_sentinel == i);
+    STATIC_ASSERT(noexcept(default_sentinel == i));
+    assert(!(i != default_sentinel));
+    STATIC_ASSERT(noexcept(!(i != default_sentinel)));
+    assert(!(default_sentinel != i));
+    STATIC_ASSERT(noexcept(!(default_sentinel != i)));
+#endif // __cpp_lib_concepts
+}
+
+template <class T, class CharT = char, class Traits = char_traits<CharT>>
+void test_ostream(basic_ostream<CharT, Traits>& os) {
+    using I = ostream_iterator<T, CharT, Traits>;
+
+    // Also tests strengthened noexcept on some pre-existing operations
+    I i{os};
+    STATIC_ASSERT(is_nothrow_constructible_v<I, basic_ostream<CharT, Traits>&>);
+    STATIC_ASSERT(is_nothrow_constructible_v<I, basic_ostream<CharT, Traits>&, const CharT*>);
+    STATIC_ASSERT(noexcept(*i));
+    STATIC_ASSERT(noexcept(++i));
+    STATIC_ASSERT(noexcept(i++));
+
+#ifdef __cpp_lib_concepts
+    STATIC_ASSERT(output_iterator<I, const T&>);
+
+    STATIC_ASSERT(is_same_v<typename I::difference_type, ptrdiff_t>);
+    { [[maybe_unused]] constexpr I constexprConstructed{}; }
+    STATIC_ASSERT(is_nothrow_default_constructible_v<I>);
+#endif // __cpp_lib_concepts
+}
+
+template <class CharT = char, class Traits = char_traits<CharT>>
+void test_istreambuf() {
+#ifdef __cpp_lib_concepts
+    using I = istreambuf_iterator<CharT, Traits>;
+    STATIC_ASSERT(input_iterator<I>);
+
+    constexpr I i{default_sentinel};
+    assert(i == I{});
+    assert(!(i != I{}));
+    STATIC_ASSERT(is_nothrow_constructible_v<I, default_sentinel_t>);
+
+    assert(i == default_sentinel);
+    assert(default_sentinel == i);
+    assert(!(i != default_sentinel));
+    assert(!(default_sentinel != i));
+#endif // __cpp_lib_concepts
+}
+
+template <class CharT = char, class Traits = char_traits<CharT>>
+void test_ostreambuf(basic_ostream<CharT, Traits>& os) {
+    using I = ostreambuf_iterator<CharT, Traits>;
+
+    // Also tests strengthened noexcept on some pre-existing operations
+    I i{os};
+    STATIC_ASSERT(noexcept(*i));
+    STATIC_ASSERT(noexcept(++i));
+    STATIC_ASSERT(noexcept(i++));
+
+#ifdef __cpp_lib_concepts
+    STATIC_ASSERT(output_iterator<I, const CharT&>);
+
+    STATIC_ASSERT(is_same_v<typename I::difference_type, ptrdiff_t>);
+    { [[maybe_unused]] constexpr I constexprConstructed{}; }
+    STATIC_ASSERT(is_nothrow_default_constructible_v<I>);
+#endif // __cpp_lib_concepts
+}
+
+int main() {
+    test_istream<int>();
+    test_istream<string>();
+    test_istream<int, wchar_t>();
+    test_istream<string, wchar_t>();
+
+    test_ostream<int>(cout);
+    test_ostream<string>(cout);
+    test_ostream<int>(wcout);
+    test_ostream<string>(wcout);
+
+    test_istreambuf<char>();
+    test_istreambuf<wchar_t>();
+
+    test_ostreambuf(cout);
+    test_ostreambuf(wcout);

```

Comment:
Naming nitpick: I find `test_istream` slightly confusing because you're testing `istream_iterator`, a different type. I'd recommend appending `_iterator` to these function names.

Reply:
I'm relying on the context added by the test name, which seems incredibly fragile.